adam-core/src/main/scala/org/bdgenomics/adam/converters/FastqRecordConverter.scala

```diff
@@ -143,10 +145,25 @@ class FastqRecordConverter extends Serializable with Logging {
     // get fields for first read in pair
     val readName = trimTrailingReadNumber(lines(0).drop(1))
     val readSequence = lines(1)
-    val readQualities = lines(3)
+
+    if (stringency == ValidationStringency.STRICT && lines(3) == "*" && readSequence.length > 1)
+      throw new Exception(s"Fastq quality must be defined")
+    else if (stringency == ValidationStringency.STRICT && lines(3).length != readSequence.length)
+      throw new Exception(s"Fastq sequence and quality strings must have the same length")
+
+    val readQualities =
```

Comment:
I think rather than fixing invalid FASTQ records when reading them in with non-strict validation stringency, I would prefer they get logged as invalid and dropped.  Then for strict validation stringency, encountering an invalid FASTQ record would throw an exception that causes the read operation to stop.